### DISCUSSION TITLE
Daemon mode

### DISCUSSION THREAD

**Discussion by SuperDuperDeou** [ghostty-org/ghostty#3529] [Feature Requests, Ideas] [answered]
It would work similarly to the GTK single instance mode, but without the window: Ghostty would run as a demon, and when a new Ghostty process is started, the daemon will spawn a new window and the second Ghostty process returns (unless the new Ghostty process is started with a specific CLI flag, something like `--daemon=false`, with `--gtk-single-instance-mode=false` implying it). When all the existing Ghostty windows are closed, the Ghostty daemon is still running.

This has a few applications:
- All the advantages of the GTK single instance mode, but also for the very first Ghostty window
- Spawn the quick terminal without needing a terminal window already open
- Possibility for remote control
- Might be a prerequisite for session management ( #3358 )

**Comment by jcollie** [comment, marked as answer]
See #2442

**Reply by SuperDuperDeou** [reply]
It's not quite the same thing: first, dbus activation is a platform-specific, it's at most a candidate for a platform-specific implementation, second, not ever GNU/Linux user uses Systemd to use Systemd units with, third, that still implies that the first Ghostty process opens a window, it's not having a proper daemon mode.
